### PR TITLE
F1 implementation

### PR DESCRIPTION
**Pull request by ciguaran-asapp**

### CODE REVIEW

File: flambe/learn/eval.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 from flambe.compile import Component
 from flambe.dataset import Dataset
+from .utils import select_device
```
Comment (iitzco-asapp) [inline flambe/learn/eval.py]: Lets keep consistency by always using the absolute input

File: flambe/learn/train.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
 
 from flambe.dataset import Dataset
 from flambe.compile import Schema, State, Component
+from .utils import select_device
```
Comment (iitzco-asapp) [inline flambe/learn/train.py]: Same here

File: flambe/metric/dev/binary.py [hunk 3]
```diff
@@ -244,3 +244,35 @@ def __str__(self) -> str:
         """Return the name of the Metric (for use in logging)."""
         invert_label = "Negative" if self.positive_label == 0 else "Positive"
         return f"{invert_label}{self.__class__.__name__}"
+
+
+class F1(BinaryMetric):
```
Comment (iitzco-asapp) [inline flambe/metric/dev/binary.py]: Rename it `F1Score` maybe better?

File: flambe/metric/dev/binary.py [hunk 4]
```diff
@@ -244,3 +244,35 @@ def __str__(self) -> str:
...
+        Returns
+        ------
+        torch.float
+            The computed binary metric
+
+        """
+        recall = self.recall.compute_binary(pred, target)
+        precision = self.precision.compute_binary(pred, target)
+
+        return 2 * precision * recall / (precision + recall + 1e-10)
```
Comment (iitzco-asapp) [inline flambe/metric/dev/binary.py]: Nice implementation reusing what we have. Is this the best way to handle `0` division?

File: tests/unit/metrics/test_dev.py [hunk 5]
```diff
@@ -141,34 +142,39 @@ def test_binary_recall():
     metric_test_case(y_pred, torch.tensor([1, 0, 1, 0]), BinaryRecall(0.8), 0)
 
 
-def test_inverted_binary_recall():
+@pytest.mark.parametrize("true_classes, metric, result",
```
Comment (iitzco-asapp) [inline tests/unit/metrics/test_dev.py]: 😎

**Comment by iitzco-asapp** [comment]
Left some comments.

File: flambe/metric/dev/binary.py (hunk 3 above)
Comment (jeremyasapp) [reply flambe/metric/dev/binary.py]: I like F1 personally

File: flambe/metric/dev/binary.py (hunk 4 above)
Comment (jeremyasapp) [reply flambe/metric/dev/binary.py]: I suggest using an `eps` argument to the object with a default of anything between 1e-6 and 1e-10, and use `self.eps` here so it's controllable

File: flambe/learn/eval.py [hunk 6]
```diff
@@ -51,11 +52,7 @@ def __init__(self,
         self.eval_metric = None
         self.dataset = dataset
 
-        # Select right device
-        if device is not None:
-            self.device = device
-        else:
-            self.device = "cuda" if torch.cuda.is_available() else "cpu"
+        self.device = select_device(device)
```
Comment (jeremyasapp) [inline flambe/learn/eval.py]: nice 👍

File: flambe/metric/dev/binary.py [hunk 7]
```diff
@@ -244,3 +244,35 @@ def __str__(self) -> str:
...
+    def __init__(self, threshold: float = 0.5, positive_label: int = 1):
+        super().__init__(threshold)
+        self.recall = BinaryRecall(threshold, positive_label)
+        self.precision = BinaryPrecision(threshold, positive_label)
+
+    def compute_binary(self,
+                       pred: torch.Tensor,
+                       target: torch.Tensor) -> torch.Tensor:
+        """Compute F1. Score, the harmonic mean between precision and
+        recall.
```
Comment (jeremyasapp) [inline flambe/metric/dev/binary.py]: line jump here

File: flambe/learn/utils.py [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import torch
+
+
+def select_device(device):
```
Comment (iitzco-asapp) [inline flambe/learn/utils.py]: Lets add the return statement and docs to this method.
